google/cloud/internal/debug_string.h

```diff
@@ -15,15 +15,65 @@
 #ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_INTERNAL_DEBUG_STRING_H
 #define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_INTERNAL_DEBUG_STRING_H
 
+#include "google/cloud/internal/absl_str_cat_quiet.h"
 #include "google/cloud/tracing_options.h"
 #include "google/cloud/version.h"
+#include "absl/strings/string_view.h"
 #include <string>
 
 namespace google {
 namespace cloud {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 namespace internal {
 
+/**
+ * Build strings for use in request/response logging.
+ *
+ * The intent is to produce strings with a format similar to those returned
+ * by `DebugString(google::protobuf::Message const&, TracingOptions const&)`
+ * from `google/cloud/internal/debug_string_protobuf.h` for a proto message.
+ */
+class DebugFormatter {
```

Comment:
I think we might want some easy way to compose `DebugFormatter`s for sub-messages, but that can wait for another day.